packages/frontend/src/components/DataViz/store/cartesianChartBaseSlice.ts

```diff
@@ -459,15 +450,15 @@ export const cartesianChartConfigSlice = createSlice({
             display.yAxis = display.yAxis || [];
             display.series = display.series || {};
 
-            if (fieldConfig?.y.length === 1) {
-                const yReference = fieldConfig?.y[0].reference;
-                if (yReference) {
-                    display.series[yReference] = {
-                        ...display.series[yReference],
-                        yAxisIndex: 0,
-                        valueLabelPosition: action.payload.valueLabelPosition,
-                    };
-                }
+            if (
+                action.payload.index !== undefined &&
+                action.payload.reference !== undefined
+            ) {
+                display.series[action.payload.reference] = {
+                    ...display.series[action.payload.reference],
+                    yAxisIndex: action.payload.index, // TODO: Why set the index here?
```

Comment:
so there's a way to reference the axis the series belongs to.

Reply:
Without having to .find it - perf issue that is happening on current explorer

Reply:
My question is more: why set it again when we are setting these other properties? presumably it gets set when the series is created, and this whole if block could be

`display.series[action.payload.reference].valuePosition = action.payload.valueLabelPosition`

No?